src/jquery.uls.core.js

```diff
@@ -61,6 +61,15 @@
 				// Language is unknown to ULS.
 				delete this.languages[ code ];
 			}
+
+			// Check if the language code is a redirect
+			var target = $.uls.data.isRedirect( code );
+
+			if ( target ) {
+				// Update target in the languages object and delete redirect
+				this.languages[ target ] = this.languages[ code ];
```

Comment:
What happens when the `languages` list has the original language (`ady`) and a related redirect (`ady-cyrl`)? Should we avoid this assignment in such cases?